src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,423 @@
+import net.snowflake.client.core.HttpUtil;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(TestCategoryOthers.class)
+public class ProxyLatestIT {
+
+  private static final Logger logger = Logger.getLogger(ProxyLatestIT.class.getName());
+
+  private static final String WIREMOCK_HOME_DIR = ".wiremock";
+  private static final String WIREMOCK_FILE_NAME = "wiremock-standalone-3.8.0.jar";
+  public static final String WIREMOCK_STANDALONE_URL =
+      "https://repo1.maven.org/maven2/org/wiremock/wiremock-standalone/3.8.0/wiremock-standalone-3.8.0.jar";
+  private static final String WIREMOCK_HOST = "localhost";
+  private static final String TRUST_STORE_PROPERTY = "javax.net.ssl.trustStore";
+  private static int httpProxyPort;
+  private static int httpsProxyPort;
+  private static String originalTrustStorePath;
+  private static Process wiremockStandalone;
+
+  @BeforeClass
+  public static void setUpClass() {
+    assumeFalse(RunningNotOnJava8.isRunningOnJava8());
+    assumeFalse(RunningNotOnJava21.isRunningOnJava21());
+    assumeFalse(
+        RunningNotOnGithubActionsMac
+            .isRunningOnGithubActionsMac()); // disabled until issue with access to localhost is
+    // fixed on github actions mac image
+    originalTrustStorePath = systemGetProperty(TRUST_STORE_PROPERTY);
+    downloadWiremock();
+    startWiremockStandAlone();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    System.setProperty(
+        TRUST_STORE_PROPERTY, getResourceURL("wiremock" + File.separator + "ca-cert.jks"));
+  }
+
+  @After
+  public void tearDown() {
+    unsetJvmProperties();
+    resetWiremock();
+    HttpUtil.httpClient.clear();
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    stopWiremockStandAlone();
+    restoreTrustStorePathProperty();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenSetInProperties() throws SQLException {
+    Properties props = getProperties();
+    addProxyProperties(props);
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenSetInJVMParams() throws SQLException {
+    Properties props = getProperties();
+    setJvmProperties(props);
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyNotUsedWhenNonProxyHostsMatchingInProperties() throws SQLException {
+    restoreTrustStorePathProperty();
```

Comment:
in before we setup trust store to reset it here? maybe just after the test in finally we should restore the trust store?

Reply:
I changed to execute it inside tests that needs that